server/server.go

```diff
@@ -996,13 +999,17 @@ func (s *Server) startPrometheusServer(listenAddr *net.TCPAddr) *http.Server {
 		ReadHeaderTimeout: 60 * time.Second,
 	}
 
-	go func() {
-		s.logger.Info("Prometheus server started", "addr=", listenAddr.String())
-
-		if err := srv.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+	s.logger.Info("Prometheus server started", "addr=", listenAddr.String())
+	go common.RetryForever(context.Background(), time.Second, func(context.Context) error {
+		if err := srv.ListenAndServe(); err != nil {
+			if errors.Is(err, http.ErrServerClosed) {
+				return nil
```

Comment:
@goran-ethernal I updated this - can you please double check this was the intention before my changes

Reply:
Looks good.